### ISSUE TITLE
no-new-object shouldn't complain when a null prototype is used.

### ISSUE THREAD

**Issue by RedHatter** [eslint/eslint#11133]
The `no-new-object` rule currently complains when you create a new object with a null prototype. This is incorrect behavior since an object with a null prototype cannot be create with an object literal.

**What rule do you want to change?**
no-new-object

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
I would think new default behavior, but a new option would work if keeping backwards compatibility is important.

**Please provide some example code that this change will affect:**

```js
let obj = new Object(null);
```

**What does the rule currently do for this code?**
It complains that the `The object literal notation {} is preferrable`.

**What will the rule do after it's changed?**
Do nothing.

**Are you willing to submit a pull request to implement this change?**
Yes.

**Comment by ljharb** [comment]
Yes it should, because `new Object(null)` is no different than `new Object`. Perhaps you're thinking of `Object.create(null)`? `new Object` should never ever be used for any reason period, forever.

**Comment by ljharb** [comment]
Also, yes, you can create a null object with a literal like this:
```js
const nullObject = {
  __proto__: null
};
```

**Comment by RedHatter** [comment]
You're right. I'm sorry, I don't know how I got that mixed up. I realized my mistake and was just coming here to close the issue, but you beat me to it.
